tests/manual/events/PlatformPreferencesTest.java

```diff
@@ -0,0 +1,113 @@
+import javafx.scene.layout.BorderPane;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Stream;
+import java.util.stream.Collectors;
+
+public class PlatformPreferencesTest extends Application {
+
+    private Map<String, Object> cachedPreferences;
+
+    @Override
+    public void start(Stage stage) {
+        var passButton = new Button("Pass");
+        passButton.setOnAction(e -> Platform.exit());
+
+        var failButton = new Button("Fail");
+        failButton.setOnAction(e -> {
+            Platform.exit();
+            throw new AssertionError("Platform preferences were not correctly reported");
+        });
+
+        var textArea = new TextArea();
+        textArea.setEditable(false);
+
+        var clearButton = new Button("Clear Log");
+        clearButton.setOnAction(e -> textArea.setText(""));
+
+        var box = new VBox();
+        box.setSpacing(20);
+        box.getChildren().add(new VBox(10,
+            new VBox(
+                new Label("1. On a supported platform, change any of the platform preferences."),
+                new Label("    See javafx.application.Platform.Preferences for a list of supported platforms.")),
+            new Label("2. Observe whether the changed preferences are reported in the log below."),
+            new Label("3. Click \"Pass\" if the changes were correctly reported, otherwise click \"Fail\"."),
+            new HBox(5, passButton, failButton, clearButton)
+        ));
+
+        var root = new BorderPane();
+        root.setPadding(new Insets(20));
+        root.setTop(box);
+        root.setCenter(textArea);
+        BorderPane.setMargin(textArea, new Insets(20, 0, 0, 0));
+
+        cachedPreferences = new HashMap<>(Platform.getPreferences());
+        textArea.setText("preferences = " + formatPrefs(cachedPreferences.entrySet().stream()));
+
+        Platform.getPreferences().addListener(
+            (InvalidationListener)observable -> {
+                Stream<Map.Entry<String, Object>> changed = Platform.getPreferences().entrySet().stream()
+                        .filter(entry -> !Objects.equals(entry.getValue(), cachedPreferences.get(entry.getKey())));
+
+                double scrollTop = textArea.getScrollTop();
+                textArea.setText(textArea.getText() + "changed = " + formatPrefs(changed));
+                textArea.setScrollTop(scrollTop);
+
+                cachedPreferences = new HashMap<>(Platform.getPreferences());
+            });
+
+        stage.setScene(new Scene(root));
+        stage.show();
+    }
+
+    public static void main(String[] args) {
+        Application.launch(args);
+    }
+
+    private static String formatPrefs(Stream<Map.Entry<String, Object>> prefs) {
+        String entries = prefs
+                .sorted(Map.Entry.comparingByKey())
+                .map(Object::toString)
```

Comment:
very, very minor: you could format arrays, to avoid [Ljavafx.scene.paint.Color;@2bacc543

Reply:
Done.